presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1659,13 +1666,19 @@ private void undoAlterPartitionOperations()
 
         private void undoUpdateStatisticsOperations()
         {
+            ImmutableList.Builder<CompletableFuture<?>> undoUpdateFutures = ImmutableList.builder();
             for (UpdateStatisticsOperation operation : updateStatisticsOperations) {
-                try {
-                    operation.undo(delegate);
-                }
-                catch (Throwable throwable) {
-                    logCleanupFailure(throwable, "failed to rollback: %s", operation.getDescription());
-                }
+                undoUpdateFutures.add(CompletableFuture.runAsync(() -> {
+                    try {
+                        operation.undo(delegate);
+                    }
+                    catch (Throwable throwable) {
+                        logCleanupFailure(throwable, "failed to rollback: %s", operation.getDescription());
+                    }
+                }, updateExecutor));
+            }
+            for (CompletableFuture<?> undoUpdateFuture : undoUpdateFutures.build()) {
+                MoreFutures.getFutureValue(undoUpdateFuture, PrestoException.class);
```

Comment:
just use `getFutureValue(undoUpdateFuture)`. none of the futures can fail as we catch all throwables. You could add a comment also.

Reply:
done